starknet/src/utils/space_manager.cairo

```diff
@@ -0,0 +1,62 @@
+#[starknet::contract]
+mod SpaceManager {
+    use starknet::{ContractAddress, info};
+
+    #[storage]
+    struct Storage {
+        _spaces: LegacyMap::<ContractAddress, bool>
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        SpaceEnabled: SpaceEnabled,
+        SpaceDisabled: SpaceDisabled
+    }
+
+    #[derive(Drop, PartialEq, starknet::Event)]
+    struct SpaceEnabled {
+        space: ContractAddress
+    }
+
+    #[derive(Drop, PartialEq, starknet::Event)]
+    struct SpaceDisabled {
+        space: ContractAddress
+    }
+
+    #[generate_trait]
+    impl InternalImpl of InternalTrait {
+        fn initializer(ref self: ContractState, mut spaces: Span<ContractAddress>) {
+            loop {
+                match spaces.pop_front() {
+                    Option::Some(space) => {
+                        self._spaces.write(*space, true);
```

Comment:
we do not check if `space.is_zero()` here, but we do in `enable_space`. thoughts?

Reply:
I kept it consistent with [sx-evm](https://github.com/snapshot-labs/sx-evm/blob/a17a4abbc8a35276163fbbb051bf14c879980b96/src/utils/SpaceManager.sol#L26) but that doesnt really make much sense. Think we should add the zero check.

Reply:
added the zero (and also duplicate) check, along with regression tests